hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/InternalExecutionService.java

```diff
@@ -18,14 +18,27 @@
 
 import com.hazelcast.spi.ExecutionService;
 
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
 /**
  * The internal {@link ExecutionService}. Currently this method doesn't contains any additional methods, but
  * in the future they can be added here.
  */
 public interface InternalExecutionService extends ExecutionService {
 
+    ExecutorService getDurable(String name, boolean scheduled);
 
     void executeDurable(String name, Runnable command);
 
+    ScheduledFuture<?> scheduleDurable(String name, Runnable command, long delay, TimeUnit unit);
+
+    <V> ScheduledFuture<V> scheduleDurable(String name, Callable<V> command, long delay, TimeUnit unit);
+
+    ScheduledFuture<?> scheduleDurableWithRepetition(String name, Runnable command,
```

Comment:
AFAIK default queue capacity is integer max for now and it is not configurable for this scheduled executor. So at least memory leaks can be seen if there is a slow db interaction for a task for example. Not sure what to do here. First thing comes, why do we have this `scheduleDurableWithRepetition` method instead of regular `scheduleAtFixedRate` or `scheduleWithFixedDelay` as in `java.util.concurrent.ScheduledExecutorService`? Maybe we can completely remove it or fix its implementation because this doesn't provide similar expected behaviour with `java.util.concurrent.ScheduledExecutorService` from this very perspective.